addons/stock/report/report_stock_rule.py

```diff
@@ -0,0 +1,96 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, models, _
+from odoo.exceptions import UserError
+
+
+class ReportStockRule(models.AbstractModel):
+    _name = 'report.stock.report_stock_rule'
+
+    @api.model
+    def _get_report_values(self, docids, data=None):
+        product = self.env['product.product'].browse(data['product_id'])
+        warehouses = self.env['stock.warehouse'].browse(data['warehouse_ids'])
+
+        routes = self._get_routes(data)
+
+        rules = routes.mapped('rule_ids').filtered(lambda r: r.warehouse_id == False or r.warehouse_id in (warehouses))
+        rules_and_loc = self._get_rules_and_loc(rules, product)
+        all_src = self.env['stock.location'].concat(*([r['source'] for r in rules_and_loc]))
+        all_dest = self.env['stock.location'].concat(*([r['destination'] for r in rules_and_loc]))
+        locations = all_src + all_dest
+        start_locations = all_src - all_dest
+        ordered_locations = self.env['stock.location']
+        locations = self._sort_locations(rules_and_loc, start_locations, ordered_locations)
+        locations_names = [location.display_name for location in locations]
+        route_lines = []
+        if locations.mapped('putaway_strategy_id'):
+            putaway_line = []
+            for x in range(len(locations_names)):
+                putaway_line.append([])
+            for location in locations:
+                if location.putaway_strategy_id:
+                    putaway = location.putaway_strategy_id._get_putaway_rule(product)
+                    idx = locations_names.index(location.display_name)
+                    tpl = (putaway, location, False, )
+                    putaway_line[idx] = tpl
+            route_lines.append(putaway_line)
+        route_colors = []
+        colors = ['#FFA500', '#800080', '#228B22', '#008B8B', '#4682B4', '#FF0000', '#32CD32']
```

Comment:
we should probably make an helper so that partners can change the colours + we'll be able to improve it easily